base/special/j1f.jl

```diff
@@ -0,0 +1,320 @@
+cos_unchecked(x::Float32) = ccall((:cosf, libm), Float32, (Float32,), x)
+log_unchecked(x::Float32) = ccall((:logf, libm), Float32, (Float32,), x)
+
+# For x >= 8, the asymptotic expansions of pone is
+#     1 + 15 / 128 s^2 - 4725 / 2^15 s^4 - ..., where s = 1/x.
+# We approximate pone by
+#     pone(x) = 1 + R / S
+# where  R = pr0 + pr1 * s^2 + pr2 * s^4 + ... + pr5 * s^10
+#     S = 1 + ps0 * s^2 + ... + ps4 * s^10
+# and
+#     |pone(x) - 1 - R / S| <= 2^-60.06
+
+# for x in [inf, 8] = 1 / [0, 0.125]
+const pr8 = (0.0000000000f+00, # 0x00000000
+             1.1718750000f-01, # 0x3df00000
+             1.3239480972f+01, # 0x4153d4ea
+             4.1205184937f+02, # 0x43ce06a3
+             3.8747453613f+03, # 0x45722bed
+             7.9144794922f+03, # 0x45f753d6
+             )::NTuple{6,Float32}
+const ps8 = (1.1420736694f+02, # 0x42e46a2c
+             3.6509309082f+03, # 0x45642ee5
+             3.6956207031f+04, # 0x47105c35
+             9.7602796875f+04, # 0x47bea166
+             3.0804271484f+04, # 0x46f0a88b
+             )::NTuple{5,Float32}
+# for x in [8, 4.5454] = 1 / [0.125, 0.22001]
+const pr5 = (1.3199052094f-11, # 0x2d68333f
+             1.1718749255f-01, # 0x3defffff
+             6.8027510643f+00, # 0x40d9b023
+             1.0830818176f+02, # 0x42d89dca
+             5.1763616943f+02, # 0x440168b7
+             5.2871520996f+02, # 0x44042dc6
+             )::NTuple{6,Float32}
+const ps5 = (5.9280597687f+01, # 0x426d1f55
+             9.9140142822f+02, # 0x4477d9b1
+             5.3532670898f+03, # 0x45a74a23
+             7.8446904297f+03, # 0x45f52586
+             1.5040468750f+03, # 0x44bc0180
+             )::NTuple{5,Float32}
+const pr3 = (3.0250391081f-09, # 0x314fe10d
+             1.1718686670f-01, # 0x3defffab
+             3.9329774380f+00, # 0x407bb5e7
+             3.5119403839f+01, # 0x420c7a45
+             9.1055007935f+01, # 0x42b61c2a
+             4.8559066772f+01, # 0x42423c7c
+             )::NTuple{6,Float32}
+const ps3 = (3.4791309357f+01, # 0x420b2a4d
+             3.3676245117f+02, # 0x43a86198
+             1.0468714600f+03, # 0x4482dbe3
+             8.9081134033f+02, # 0x445eb3ed
+             1.0378793335f+02, # 0x42cf936c
+             )::NTuple{5,Float32}
+# for x in [2.8570, 2] = 1 / [0.3499, 0.5]
+const pr2 = (1.0771083225f-07, # 0x33e74ea8
+             1.1717621982f-01, # 0x3deffa16
+             2.3685150146f+00, # 0x401795c0
+             1.2242610931f+01, # 0x4143e1bc
+             1.7693971634f+01, # 0x418d8d41
+             5.0735230446f+00, # 0x40a25a4d
+             )::NTuple{6,Float32}
+const ps2 = (2.1436485291f+01, # 0x41ab7dec
+             1.2529022980f+02, # 0x42fa9499
+             2.3227647400f+02, # 0x436846c7
+             1.1767937469f+02, # 0x42eb5bd7
+             8.3646392822f+00, # 0x4105d590
+             )::NTuple{5,Float32}
+
+function ponef(x::Float32)
+    ix::Int32 = reinterpret(Int32, x)
+    ix &= 0x7fffffff
+    if ix >= 0x41000000
+        p = pr8
```

Comment:
For the simple ones, I didn't inline them mainly because it's a relatively error prone process as I'm translating the syntax. The compiler is also smart enough to inline the constant so that shouldn't be a big issue (when the constant doesn't depend on the input)